docs/developer-docs/integrations/independently-verifying-ic-signatures.md

```diff
@@ -0,0 +1,125 @@
+
+- `delegation` (`map`): A map containing the fields: 
+
+    - `pubkey` (`blob`): A public key.
+    
+    - `expiration` (`nat`): The delegation's expiration, defined in nanoseconds analogously to the `ingress_expiry`. 
+    
+    - `targets` (`array` of `CanisterId`, optional): Sets the delegation to apply only for requests sent to the canisters within the canister list; has a maximum of 1000 canisters.
+
+    - `senders` (`array` of `Principal`, optional): Sets the delegation to only apply for requests originating from the principals in the list.
+    
+    - `signature` (`blob`): The signature on the `32-byte` delegation field map, using the 27 bytes `\x1Aic-request-auth-delegation` as the domain separator.
+
+The first delegation in the array has a signature created using the public key corresponding with the `sender_pubkey` field. All subsequent delegations are signed with the public key corresponding to the key contained in the preceding delegation.
+
+The `sender_sig` field is calculated by signing concatenating the domain separator, `\x0Aic-request`, which is 11 bytes, with the 32 byte request ID, using the secret key that belongs to the key specified in the last delegation. If no delegations are present, the public key specified in `sender_pubkey` is used.  If the delegation field is present, it should contain no more than 20 delegations.
+
+When a canister call is executed on the IC, the signature of the call is validated automatically. As an additional layer of validation, these calls can be verified independently through a Rust crate. 
+
+## Verifying signatures with agent
+
+To verify a signature with an agent, an accepted identity is required. The following are the accepted types of signatures in identities:
+
+- Ed25519 and ECDSA signatures.
+
+    - Plain signatures are supported for the schemes.
+
+- Ed25519 or ECDSA on curve P-256 (also known as secp256r1).
+
+    - Using SHA-256 as a hash function.
+
+    - Using the Koblitz curve in secp256k1.
+
+When these identities are encoded as a principal, an agent will attach a suffix byte which indicates whether the identity is anonymous or self-authenticating. 
+
+Self-authenticating identities using an Ed25519 or ECDSA curve will have a suffix of `2`, while an anonymous identity has a single byte of `4`. 
+
+
+## Verifying signatures with the Rust `ic-validator-ingress-message` crate
+
+The Rust [ic-validator-ingress-message](https://github.com/dfinity/ic/tree/master/rs/validator) crate has been developed specifically for validating message signatures. Within this crate, the `IngressMessageVerifier` class can be used. Within the class, the `InternetIdentityAuthResponse` can be validated as a whole using the Delegation chain. 
+
+The following example displays an example verifier for ingress messages. This example uses a hard-coded root of trust that is set to be the NNS root public key and uses system time to derive the current time. 
+
+```rust
```

Comment:
As a follow up, it would be nice to link to the examples repo with a little project set up for this!